File: cmd/milestone.go
package cmd

import (
	"github.com/eugene-software/assembla-cli/internal"
	"github.com/spf13/cobra"
)

var milestoneCmd = &cobra.Command{
	Use:   "milestone",
	Short: "Manage milestones",
}

var milestoneListCmd = &cobra.Command{
	Use:   "list",
	Short: "List milestones",
	RunE: func(cmd *cobra.Command, args []string) error {
		space := requireSpace()
		asJSON, _ := cmd.Flags().GetBool("json")
		showAll, _ := cmd.Flags().GetBool("all")
		completed, _ := cmd.Flags().GetBool("completed")

		var path string
		if showAll {
			path = internal.APIPath("spaces", space, "milestones", "all")
		} else if completed {
			path = internal.APIPath("spaces", space, "milestones", "completed")
		} else {
			path = internal.APIPath("spaces", space, "milestones", "upcoming")
		}

		data, err := Client.Get(path, nil)
		if err != nil {
			return err
		}

		if asJSON {
			internal.PrintJSON(data)
		} else {
			internal.PrintTable(
				internal.ToSlice(data),
				[]string{"id", "title", "due_date", "is_completed", "planner_type"},
				[]string{"ID", "Title", "Due Date", "Completed", "Type"},
			)
		}
		return nil
	},
}

var milestoneShowCmd = &cobra.Command{
	Use:   "show <milestone_id>",
	Short: "Show milestone details",
	Args:  cobra.ExactArgs(1),
	RunE: func(cmd *cobra.Command, args []string) error {
		space := requireSpace()
		asJSON, _ := cmd.Flags().GetBool("json")
		milestoneID := args[0]

		data, err := Client.Get(internal.APIPath("spaces", space, "milestones", milestoneID), nil)
		if err != nil {
			return err
		}

		if asJSON {
			internal.PrintJSON(data)
		} else {
			internal.PrintDetail(data, []string{
				"id", "title", "description", "due_date",
				"is_completed", "planner_type", "created_at", "updated_at",
			})
		}
		return nil
	},
}

func init() {
	milestoneListCmd.Flags().Bool("all", false, "Show all milestones")
	milestoneListCmd.Flags().Bool("completed", false, "Show completed milestones")
	milestoneListCmd.Flags().Bool("json", false, "Output as JSON")

	milestoneShowCmd.Flags().Bool("json", false, "Output as JSON")

	milestoneCmd.AddCommand(milestoneListCmd)
	milestoneCmd.AddCommand(milestoneShowCmd)
}

File: assembla_cli/commands/milestone.py
from __future__ import annotations

import click

from assembla_cli.formatting import print_detail, print_json, print_table


@click.group()
def milestone():
    """Manage milestones."""


@milestone.command("list")
@click.option("--all", "show_all", is_flag=True, help="Show all milestones (not just upcoming)")
@click.option("--completed", is_flag=True, help="Show completed milestones")
@click.option("--json", "as_json", is_flag=True, help="Output as JSON")
@click.pass_context
def milestone_list(ctx, show_all, completed, as_json):
    """List milestones in the space."""
    client = ctx.obj["client"]
    space = ctx.obj["space"]
    if show_all:
        path = f"/spaces/{space}/milestones/all"
    elif completed:
        path = f"/spaces/{space}/milestones/completed"
    else:
        path = f"/spaces/{space}/milestones/upcoming"
    data = client.get(path)
    if as_json:
        print_json(data)
    else:
        print_table(
            data or [],
            columns=["id", "title", "due_date", "is_completed", "planner_type"],
            headers=["ID", "Title", "Due Date", "Completed", "Type"],
        )


@milestone.command("show")
@click.argument("milestone_id")
@click.option("--json", "as_json", is_flag=True, help="Output as JSON")
@click.pass_context
def milestone_show(ctx, milestone_id, as_json):
    """Show milestone details."""
    client = ctx.obj["client"]
    space = ctx.obj["space"]
    data = client.get(f"/spaces/{space}/milestones/{milestone_id}")
    if as_json:
        print_json(data)
    else:
        print_detail(data, [
            "id", "title", "description", "due_date",
            "is_completed", "planner_type", "created_at", "updated_at",
        ])
